workload/src/jwt_watcher.c

```diff
@@ -0,0 +1,223 @@
+#include "jwt_watcher.h"
+#include "client.h"
+
+// Function that will run on thread spun for watcher
+int workloadapi_JWTWatcher_JWTbackgroundFunc(void *_watcher)
+{
+    workloadapi_JWTWatcher *watcher = (workloadapi_JWTWatcher *) _watcher;
+
+    err_t error = NO_ERROR;
+    do {
+        // TODO: CHECK ERROR AND RETRY
+        error = workloadapi_Client_WatchJWTBundles(watcher->client, watcher);
+    } while(error != ERROR5); // error5 == client closed
+    return (int) error;
+}
+
+// new watcher, creates client if not provided.
+workloadapi_JWTWatcher *
+workloadapi_newJWTWatcher(workloadapi_JWTWatcherConfig config,
+                       workloadapi_JWTCallback jwt_callback, err_t *error)
+{
+
+    workloadapi_JWTWatcher *newW
+        = (workloadapi_JWTWatcher *) calloc(1, sizeof *newW);
+
+    // set by calloc:
+    // newW->updated = false;
+    // newW->close_error = NO_ERROR;
+    // newW->update_error = NO_ERROR;
+    // newW->thread_error = thrd_success;
+
+    newW->closed = true;
+
+    if(config.client) {
+        newW->client = config.client;
+        newW->owns_client = false;
+        if(config.client_options) {
+            for(int i = 0; i < arrlen(config.client_options); i++) {
```

Comment:
Fazer "for(size_t i = 0, size = arrlenu(config.client_options); i < size; ++i)" ao invés de "for(int i = 0; i < arrlen(config.client_options); i++)". Razão: menos custoso.